internal/resource/config/oauth/oauth_auth_server_settings_resource.go

```diff
@@ -0,0 +1,1167 @@
+	}
+
+	if internaltypes.IsDefined(plan.PendingAuthorizationTimeout) {
+		addRequest.PendingAuthorizationTimeout = plan.PendingAuthorizationTimeout.ValueInt64()
+	}
+
+	if internaltypes.IsDefined(plan.DevicePollingInterval) {
+		addRequest.DevicePollingInterval = plan.DevicePollingInterval.ValueInt64()
+	}
+
+	if internaltypes.IsDefined(plan.ActivationCodeCheckMode) {
+		addRequest.ActivationCodeCheckMode = plan.ActivationCodeCheckMode.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.BypassActivationCodeConfirmation) {
+		addRequest.BypassActivationCodeConfirmation = plan.BypassActivationCodeConfirmation.ValueBool()
+	}
+
+	if internaltypes.IsDefined(plan.UserAuthorizationConsentPageSetting) {
+		addRequest.UserAuthorizationConsentPageSetting = plan.UserAuthorizationConsentPageSetting.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.UserAuthorizationConsentAdapter) {
+		addRequest.UserAuthorizationConsentAdapter = plan.UserAuthorizationConsentAdapter.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ApprovedScopesAttribute) {
+		addRequest.ApprovedScopesAttribute = plan.ApprovedScopesAttribute.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ApprovedAuthorizationDetailAttribute) {
+		addRequest.ApprovedAuthorizationDetailAttribute = plan.ApprovedAuthorizationDetailAttribute.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ParReferenceTimeout) {
+		addRequest.ParReferenceTimeout = plan.ParReferenceTimeout.ValueInt64Pointer()
+	}
+
+	if internaltypes.IsDefined(plan.ParReferenceLength) {
+		addRequest.ParReferenceLength = plan.ParReferenceLength.ValueInt64Pointer()
+	}
+
+	if internaltypes.IsDefined(plan.ParStatus) {
+		addRequest.ParStatus = plan.ParStatus.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ClientSecretRetentionPeriod) {
+		addRequest.ClientSecretRetentionPeriod = plan.ClientSecretRetentionPeriod.ValueInt64Pointer()
+	}
+
+	if internaltypes.IsDefined(plan.JwtSecuredAuthorizationResponseModeLifetime) {
+		addRequest.JwtSecuredAuthorizationResponseModeLifetime = plan.JwtSecuredAuthorizationResponseModeLifetime.ValueInt64Pointer()
+	}
+
+	return nil
+
+}
+
+// Metadata returns the resource type name.
+func (r *oauthAuthServerSettingsResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_oauth_auth_server_settings"
+}
+
+func (r *oauthAuthServerSettingsResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readOauthAuthServerSettingsResponse(ctx context.Context, r *client.AuthorizationServerSettings, state *oauthAuthServerSettingsResourceModel) {
+	state.Id = types.StringValue("id")
```

Comment:
Is there a unique value in the response that would be usable here? Ideally we can have different ids for different resources of the same type.

Reply:
Nevermind I see this is an edit-only resource based on the comment on the delete method.